.gitlab/binary_build/system_probe.yml

```diff
@@ -10,6 +10,10 @@
     # TODO: remove this once we switch to k8s runners, they won't have this problem
     - find "$CI_BUILDS_DIR" ! -path '*DataDog/datadog-agent*' -depth  # -delete implies -depth
     - find "$CI_BUILDS_DIR" ! -path '*DataDog/datadog-agent*' -delete || true  # Allow failure, we can't remove parent folders of datadog-agent
+    # XXX: system-probe images does not run `entrypoint.sh` which runs `source /root/.bashrc`.
+    # Since the path of Go binary is added into PATH in the bashrc, we need to run it at some point.
+    # This is a workaround until I understand why we don't wanna run `entrypoint.sh` for system-probe. 
+    - source /root/.bashrc
```

Comment:
FWIW `kernel_matrix_testing_system_probe` and `kernel_matrix_testing_security_agent` stages failed when I sourced .bashrc (because `metal_instance` could not be resolved 🤷 ) so this might be a bit more tricky. I'll definitely need some context from the ebpf / Devx-Infra team

Reply:
We recently had some PRs from the devx team removing the sourcing of `.bashrc`:

- https://github.com/DataDog/datadog-agent/pull/29237 
- https://github.com/DataDog/datadog-agent/pull/29007

From what I understand, the new feature flag to honor entrypoint means we don't have to manually source `.bashrc` anymore. Do you have the link to that pipeline that failed? Maybe we can take a look at what was missing.

Reply:
That's my understanding as well but I don't see the entrypoint being used in `system_probe-[x64|arm64]` images ([dockerfile](https://github.com/DataDog/datadog-agent-buildimages/blob/main/system-probe_x64/Dockerfile))

Here is an [example job](https://gitlab.ddbuild.io/DataDog/datadog-agent/-/jobs/650000785) where go is not found because PATH is updated in the bashrc now. If sourcing bashrc is a problem, we maybe can set a default env var in these 2 images ?

Reply:
I asked the team, the reason we removed the entrypoint was because the k8s runners didn't support it. Now that they do, we can add the entrypoint back and so we can avoid sourcing `bashrc` everywhere. We can do it in a separate PR ourselves or you can add it to the one you already have opened on buildimages if you prefer.

Reply:
I added this in the buildimages PR, jobs seem to pass without sourcing in this repo 🙏 Thanks for the context !